src/video.rs

```diff
@@ -0,0 +1,113 @@
+use eframe::{egui, epi};
+use ini::{Ini, Properties, SectionSetter};
+
+use crate::utils::*;
+use crate::{get_ini_value, graphics::AppGraphics, int_text_box, simple_checkbox, DisplayFormat};
+
+pub struct AppVideo {
+	pub(crate) display_format: DisplayFormat,
+	pub(crate) window_size_x: i32,
+	pub(crate) window_size_y: i32,
+	pub(crate) internal_resolution_enabled: bool,
+	pub(crate) internal_resolution_x: i32,
+	pub(crate) internal_resolution_y: i32,
+}
+
+impl IniConfig for AppVideo {
+	const SECTION: &'static str = "Resolution";
+	type Error = std::io::Error;
+
+	fn read_body(resolution_section: &Properties) -> Result<Self, Self::Error> {
+		//TODO: handle the error case
+		let display_format =
+			DisplayFormat::from_u8(get_ini_value::<u8>(resolution_section, "Display"))
+				.unwrap_or_default();
+		let window_size_x = get_ini_value(resolution_section, "Width");
+		let window_size_y = get_ini_value(resolution_section, "Height");
+		let internal_resolution_enabled = get_ini_value(resolution_section, "r.Enable");
+		let internal_resolution_x = get_ini_value(resolution_section, "r.Width");
+		let internal_resolution_y = get_ini_value(resolution_section, "r.Height");
+		Ok(Self {
+			display_format,
+			window_size_x,
+			window_size_y,
+			internal_resolution_enabled,
+			internal_resolution_x,
+			internal_resolution_y,
+		})
+	}
+}
+
+impl IniConfigWriteCtx for AppVideo {
+	type Context = AppGraphics;
+
+	fn write_body<'a, 'b>(&'a self, section: &'b mut SectionSetter<'b>) {
+		section
+			.set("Display", self.display_format.to_string())
```

Comment:
This causes it to save like `Display=Windowed` instead of the expected `Display=0`
Can be fixed with something like `(self.display_format as i32).to_string()`